src/magic_folder/tahoe_client.py

```diff
@@ -125,7 +128,14 @@ class TahoeClient(object):
     """
 
     url = attr.ib(validator=attr.validators.instance_of(DecodedURL))
-    http_client = attr.ib(validator=attr.validators.instance_of(HTTPClient))
+
+    # treq should provide an interface but it doesn't ...  HTTPClient and
+    # StubTreq are both the same kind of thing.  HTTPClient is the one that
+    # does real networking, StubTreq is the one that operates in-memory on a
+    # local resource object.
```

Comment:
Well, so we also (now?) have an `HTTPClient` that does the in-memory stuff too via `magic_folder.client.create_testing_http_client`. I'm not suggesting immediate changes here -- just noting that. So e.g. we could standardize around "HTTPClient" but still do in-memory testing.

Reply:
It does seem like that should be possible, yea.  Actually, looking more closely, I don't even know what `StubTreq` exists.  Maybe it was a precursor to `RequestTraversalAgent`.